src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs

```diff
@@ -123,19 +123,38 @@ private async Task TransformProjectToContainer(Service service, ProjectRunInfo p
             serviceDescription.RunInfo = dockerRunInfo;
         }
 
-        private static string DetermineContainerImage(ProjectRunInfo project)
+        private string DetermineContainerImage(ProjectRunInfo project)
         {
+            var baseImageTag = !string.IsNullOrEmpty(project.ContainerBaseTag) ? project.ContainerBaseTag : project.TargetFrameworkVersion;
+
             string baseImage;
             if (!string.IsNullOrEmpty(project.ContainerBaseImage))
             {
                 baseImage = project.ContainerBaseImage;
             }
             else
             {
-                baseImage = project.IsAspNet ? "mcr.microsoft.com/dotnet/core/aspnet" : "mcr.microsoft.com/dotnet/core/runtime";
-            }
+                Version? baseImageVersion = null;
+                try
+                {
+                    baseImageVersion = new Version(baseImageTag);
+                }
+                catch
+                {
+                    _logger.LogInformation($"Could not determine version of docker base image for tag: {baseImageTag}");
```

Comment:
Shouldn't you still let this throw?

Reply:
Curious if we should, currently it falls back to 2.1/3.1 but I dunno if that makes sense. If we throw we should probably advise selecting an explicit base image via the option.

Reply:
Actually, that sounds better to me. I'm gonna make it throw.